### ISSUE TITLE
Pods running on AKS Windows nodes are not able to mount Azure File Shares after changing CRI to containerd

### ISSUE THREAD

**Issue by rnemeth90** [kubernetes-sigs/azurefile-csi-driver#923]
**What happened**:
Pods running on AKS Windows nodes are not able to mount Azure File Shares after changing CRI to containerd. Here is an except from a deployment

**What you expected to happen**:
Pods should mount Azure File shares using built-in CSI driver on Windows nodes running containerd

**How to reproduce it**:
Create a deployment using an inline AzureFile share, on an Windows node running containerd
Relevant portion of my deployment manifest:

```
          terminationMessagePolicy: File
          volumeMounts:
          - mountPath: 'S:'
            name: azure
      dnsPolicy: ClusterFirst
      nodeSelector:
        kubernetes.azure.com/agentpool: npwin3
        kubernetes.io/os: windows
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: dam-restapi-c667
      serviceAccountName: dam-restapi-c667
      terminationGracePeriodSeconds: 30
      tolerations:
        - effect: NoSchedule
          key: nodePool
          operator: Equal
          value: aksnpwin3
      volumes:
      - name: azure
        csi:
          driver: file.csi.azure.com
          volumeAttributes:
            secretName: dam-restapi-c667-azure-temporaryfileshare-secret
            shareName: labs
```

**Anything else we need to know?**:

Logs from the AzureFile CSI pod:

```
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:06.617239   10476 utils.go:118] GRPC call: /csi.v1.Node/NodeUnpublishVolume
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:06.617239   10476 utils.go:119] GRPC request: {"target_path":"c:\\var\\lib\\kubelet\\pods\\2dc71eb5-315a-4aa5-8e99-974a11c465ca\\volumes\\kubernetes.io~csi\\azure\\mount","volume_id":"csi-b13711ec8f9ad115cbfc1b519400beba78a48f31a245b5749b058dc7d49ac5a8"}
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:06.617476   10476 nodeserver.go:119] NodeUnpublishVolume: unmounting volume csi-b13711ec8f9ad115cbfc1b519400beba78a48f31a245b5749b058dc7d49ac5a8 on c:\var\lib\kubelet\pods\2dc71eb5-315a-4aa5-8e99-974a11c465ca\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:06.618446   10476 safe_mounter_v1beta_windows.go:80] SMBUnmount: local path: c:\var\lib\kubelet\pods\2dc71eb5-315a-4aa5-8e99-974a11c465ca\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:06.618484   10476 safe_mounter_v1beta_windows.go:107] Remove directory: c:\var\lib\kubelet\pods\2dc71eb5-315a-4aa5-8e99-974a11c465ca\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:06.619842   10476 nodeserver.go:123] NodeUnpublishVolume: unmount volume csi-b13711ec8f9ad115cbfc1b519400beba78a48f31a245b5749b058dc7d49ac5a8 on c:\var\lib\kubelet\pods\2dc71eb5-315a-4aa5-8e99-974a11c465ca\volumes\kubernetes.io~csi\azure\mount successfully
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:06.619938   10476 utils.go:125] GRPC response: {}
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.544704   10476 utils.go:118] GRPC call: /csi.v1.Node/NodePublishVolume
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.544704   10476 utils.go:119] GRPC request: {"target_path":"c:\\var\\lib\\kubelet\\pods\\a388d0b1-9711-49ec-a1c6-e047081c038d\\volumes\\kubernetes.io~csi\\azure\\mount","volume_capability":{"AccessType":{"Mount":{}},"access_mode":{"mode":1}},"volume_context":{"csi.storage.k8s.io/ephemeral":"true","csi.storage.k8s.io/pod.name":"dam-restapi-c667-5494b95f8f-9m2zg","csi.storage.k8s.io/pod.namespace":"dam-c667","csi.storage.k8s.io/pod.uid":"a388d0b1-9711-49ec-a1c6-e047081c038d","csi.storage.k8s.io/serviceAccount.name":"dam-restapi-c667","secretName":"dam-restapi-c667-azure-temporaryfileshare-secret","shareName":"labs"},"volume_id":"csi-b5970cca47532c18d0373786088e1a0b1b44ce6d87fbd0e51f5ff00b6ba3aaa7"}
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.545178   10476 nodeserver.go:63] NodePublishVolume: ephemeral volume(csi-b5970cca47532c18d0373786088e1a0b1b44ce6d87fbd0e51f5ff00b6ba3aaa7) mount on c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount, VolumeContext: map[csi.storage.k8s.io/ephemeral:true csi.storage.k8s.io/pod.name:dam-restapi-c667-5494b95f8f-9m2zg csi.storage.k8s.io/pod.namespace:dam-c667 csi.storage.k8s.io/pod.uid:a388d0b1-9711-49ec-a1c6-e047081c038d csi.storage.k8s.io/serviceAccount.name:dam-restapi-c667 getaccountkeyfromsecret:true secretName:dam-restapi-c667-azure-temporaryfileshare-secret secretnamespace:dam-c667 shareName:labs storageaccount:]
[csi-azurefile-node-win-mxrct azurefile] W0224 14:35:10.545178   10476 azurefile.go:532] parsing volumeID(csi-b5970cca47532c18d0373786088e1a0b1b44ce6d87fbd0e51f5ff00b6ba3aaa7) return with error: error parsing volume id: "csi-b5970cca47532c18d0373786088e1a0b1b44ce6d87fbd0e51f5ff00b6ba3aaa7", should at least contain two #
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.594832   10476 nodeserver.go:250] cifsMountPath(c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount) fstype() volumeID(csi-b5970cca47532c18d0373786088e1a0b1b44ce6d87fbd0e51f5ff00b6ba3aaa7) context(map[csi.storage.k8s.io/ephemeral:true csi.storage.k8s.io/pod.name:dam-restapi-c667-5494b95f8f-9m2zg csi.storage.k8s.io/pod.namespace:dam-c667 csi.storage.k8s.io/pod.uid:a388d0b1-9711-49ec-a1c6-e047081c038d csi.storage.k8s.io/serviceAccount.name:dam-restapi-c667 getaccountkeyfromsecret:true secretName:dam-restapi-c667-azure-temporaryfileshare-secret secretnamespace:dam-c667 shareName:labs storageaccount:]) mountflags([]) mountOptions([AZURE\l17us102shtmp02]) volumeMountGroup()
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.594929   10476 safe_mounter_v1beta_windows.go:138] IsLikelyNotMountPoint: c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.594929   10476 safe_mounter_v1beta_windows.go:196] Exists path: c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.597840   10476 safe_mounter_v1beta_windows.go:196] Exists path: c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.599335   10476 azure_common_windows.go:73] Removing path: c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.599335   10476 safe_mounter_v1beta_windows.go:107] Remove directory: c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.601336   10476 safe_mounter_v1beta_windows.go:47] SMBMount: remote path: \\l17us102shtmp02.file.core.windows.net\labs. local path: c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure\mount
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:10.601336   10476 safe_mounter_v1beta_windows.go:196] Exists path: c:\var\lib\kubelet\pods\a388d0b1-9711-49ec-a1c6-e047081c038d\volumes\kubernetes.io~csi\azure
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:12.133306   10476 nodeserver.go:283] volume(csi-b5970cca47532c18d0373786088e1a0b1b44ce6d87fbd0e51f5ff00b6ba3aaa7) mount "\\\\l17us102shtmp02.file.core.windows.net\\labs" on "c:\\var\\lib\\kubelet\\pods\\a388d0b1-9711-49ec-a1c6-e047081c038d\\volumes\\kubernetes.io~csi\\azure\\mount" succeeded
[csi-azurefile-node-win-mxrct azurefile] I0224 14:35:12.133306   10476 utils.go:125] GRPC response: {}
```

I noticed that nodeserver.go is returning an empty storageaccount field, and then getFileShareInfo function in azurefile.go is not able to parse the volumeId because it is invalid. 

https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/pkg/azurefile/nodeserver.go
https://github.com/kubernetes/csi-translation-lib/blob/master/plugins/azure_file.go

**Environment**:
- CSI Driver version: mcr.microsoft.com/oss/kubernetes-csi/azurefile-csi:v1.10.0
- Kubernetes version (use `kubectl version`): 1.21.7
- OS (e.g. from /etc/os-release): Windows Server 2019
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by andyzhangx** [comment]
that's actually mountPath behavior changed from docker to containerd on Windows, issue: https://github.com/containerd/containerd/issues/6589

`mountPath: 'S:'` would be mounted as `/c/S/` inside container, could you try that path?
From the logs, the azure file mount actually succeeded.

**Comment by rnemeth90** [comment]
Is the mount still available as "S:\" inside the container?

**Comment by andyzhangx** [comment]
> Is the mount still available as "S:" inside the container?

No, it would be mounted as `c:\S` inside Windows container

**Comment by brandongodby78** [comment]
@andyzhangx Hello, I am @rnemeth90 colleague. If we change our code to support containerd path as is `c:\s` and a fix comes out to make the mount path back to `s:`, will that update be pushed out forcefully, and break our code? Or would that be an opt in update, which would give us a chance to revert our code?

**Comment by andyzhangx** [comment]
> @andyzhangx Hello, I am @rnemeth90 colleague. If we change our code to support containerd path as is `c:\s` and a fix comes out to make the mount path back to `s:`, will that update be pushed out forcefully, and break our code? Or would that be an opt in update, which would give us a chance to revert our code?

@brandongodby78 I am not sure since this is a containerd issue, we are waiting for fix on https://github.com/containerd/containerd/issues/6589 first, I think you could track that issue.

**Comment by nodranael** [comment]
Hello @andyzhangx.  Looks like the fix you are waiting for has been included in [containerd v1.6.5](https://github.com/containerd/containerd/releases/tag/v1.6.5). 
I have a scenario where I need to mount an Azure File share as a drive (e.g. F:\), so I'm waiting for that as well. 
Do you have any idea of when the AKS Windows images will switch from v1.6.1 to that containerd version ?
Thanks for your help.

**Comment by andyzhangx** [comment]
> Hello @andyzhangx. Looks like the fix you are waiting for has been included in [containerd v1.6.5](https://github.com/containerd/containerd/releases/tag/v1.6.5). I have a scenario where I need to mount an Azure File share as a drive (e.g. F:), so I'm waiting for that as well. Do you have any idea of when the AKS Windows images will switch from v1.6.1 to that containerd version ? Thanks for your help.

@nodranael v1.6.5 on AKS windows would be ready in July
